src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
+            state = validate_state(state, offset, message)
             # XXX: This is necessary because of a bug in protocol buffer oneof.
             state = type(state).FromString(state.SerializeToString())
-            yield state, offset, decoded
+            yield state, offset, message
+
+    def __primed(self, consumer, state, attempts=5):
+        decoded = imap(
+            lambda (offset, msg): tuple((offset, self.codec.decode(msg.value))),
+            consumer
+        )
+
+        offset, message = next(decoded)
+
+        try:
+            validate_state(state, offset, message)
+        except InvalidEventError as e:
+            # See if we can drop the first few messages in an attempt to find
+            # the type we're looking for
+            while attempts > 0:
```

Comment:
While it works, this code makes me feel uneasy and a little dirty because I think I feel bad dropping messages on the floor like this. I'm worried there could be an error and mess things up...but I think it's okay.  Thoughts?